src/js/protocols/stm32usbdfu.js

```diff
@@ -764,7 +802,7 @@ STM32DFU_protocol.prototype.upload_procedure = function (step) {
 
             var write = function () {
                 if (bytes_flashed < self.hex.data[flashing_block].bytes) {
-                    var bytes_to_write = ((bytes_flashed + 2048) <= self.hex.data[flashing_block].bytes) ? 2048 : (self.hex.data[flashing_block].bytes - bytes_flashed);
+                    var bytes_to_write = ((bytes_flashed + self.transferSize) <= self.hex.data[flashing_block].bytes) ? self.transferSize : (self.hex.data[flashing_block].bytes - bytes_flashed);
```

Comment:
Probably a good idea to declare `transferSize` at the top of the file and initialise it to 2048 just in case.